codegen/src/api/mod.rs

```diff
@@ -371,6 +375,42 @@ impl RuntimeGenerator {
             })
         });
 
+        let outer_error_variants = self.metadata.pallets.iter().filter_map(|p| {
+            let variant_name = format_ident!("{}", p.name);
+            let mod_name = format_ident!("{}", p.name.to_string().to_snake_case());
+            let index = proc_macro2::Literal::u8_unsuffixed(p.index);
+
+            p.error.as_ref().map(|_| {
+                quote! {
+                    #[codec(index = #index)]
+                    #variant_name(#mod_name::Error),
+                }
+            })
+        });
+
+        let outer_error = quote! {
+            #default_derives
+            pub enum Error {
+                #( #outer_error_variants )*
+            }
+        };
+
+        let root_error_match_arms = self.metadata.pallets.iter().filter_map(|p| {
+            let variant_index = p.index;
```

Comment:
We should compare based on variant names and not indexes; indexes can chanbe between nodes (as pallets tend to live in different indexes on different chains), so comparing on names is more portable :)

Reply:
Ah okay, so a conversion from index to name should happen inside the `as_root_error()` function of this struct:
```pub struct ModuleError {
    metadata: Metadata,
    raw: RawModuleError,
}
```
using it's `metadata` field?

Reply:
There's a `details()` function on the ModuleError already which will give you the pallet name that the error came from, so I'd probably call that inside the `as_root_error` function and then hand the name to the `RootError` trait so then the codegen can use the name to find the correct variant that was generated.